Server/bkr/server/watchdog.py

```diff
@@ -1,66 +1,51 @@
-
-        col = myPaginateDataGrid.Column
-        fields = [col(name='job_id', getter=lambda x: x.recipe.recipeset.job.link, title="Job ID"),
-                  col(name='system_name', getter=lambda x: x.recipe.resource.link, title="System"),
-                  col(name='lab_controller', getter=lambda x: x.recipe.recipeset.lab_controller, title="Lab Controller"),
-                  col(name='task_name', getter=lambda x: x.recipetask.name_markup
-                        if x.recipetask is not None else None, title="Task Name"),
-                  col(name='kill_time', getter=lambda x: x.kill_time,
-                      title="Kill Time", options=dict(datetime=True))]
-
-        watchdog_grid = myPaginateDataGrid(fields=fields)
-        return dict(title="Watchdogs",
-                grid=watchdog_grid,
-                search_bar=None,
-                list=query)
-
-
-    # TODO: future cleanup so that the correct error message
-    # is given to the client code.
-    @identity.require(identity.in_group('admin'))
-    @cherrypy.expose
-    def extend(self, time):
-        '''Allow admins to push watchdog times out after an outage'''
-
-        watchdogs = []
-        for w in Watchdog.by_status(status=u'active'):
-            n_kill_time = w.kill_time + timedelta(seconds=time)
-            watchdogs.append("R:%s watchdog moved from %s to %s" % (
-                              w.recipe_id, w.kill_time, n_kill_time))
-            w.kill_time = n_kill_time
 
-        if watchdogs:
-            return "\n".join(watchdogs)
-        else:
-            return 'No active watchdogs found'
+from flask import jsonify, request
+from sqlalchemy.orm import joinedload
+
+from bkr.server.app import app
+from bkr.server.flask_util import admin_auth_required, read_json_request
+from bkr.server.model import Watchdog, Recipe, RecipeSet, Job, RecipeTask
+
+
+@app.route("/watchdogs/", methods=["GET"])
+def get_watchdogs():
+    # TODO: Format may change in future based on FE requests
+    # For now give them raw data and laugh. That's what backend heroes do
+    query = (
+        Watchdog.by_status(status=u"active")
+        .join(Watchdog.recipe)
+        .join(Recipe.recipeset)
+        .join(RecipeSet.job)
+        .order_by(Job.id)
+        .options(
+            joinedload(Watchdog.recipe)
+            .joinedload(Recipe.recipeset)
+            .joinedload(RecipeSet.job),
+            joinedload(Watchdog.recipe)
+            .joinedload(Recipe.recipeset)
+            .joinedload(RecipeSet.lab_controller),
+            joinedload(Watchdog.recipetask).joinedload(RecipeTask.task),
+        )
+    )
+
+    return jsonify(query)
+
+
+@app.route("/watchdogs/", methods=["POST"])
+@admin_auth_required
+def set_watchdogs():
+    """Allow admins to push watchdog times out after an outage"""
+    payload = read_json_request(request)
+    time = payload.get("time")
+
+    watchdogs = {}
+    for w in Watchdog.by_status(status=u"active"):
```

Comment:
do we still need the unicode?

Reply:
Yeah. This is still 2/3. I don't wanna screw my execution in Python 2

Reply:
ack